.github/workflows/ci.yaml

```diff
@@ -1,10 +1,15 @@
 name: Continuous Integration
+
 on: # rebuild any PRs and main branch changes
   pull_request:
   push:
     branches:
       - main
 
+concurrency:
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
```

Comment:
What's this fanciness? Didn't know they had this.

Reply:
Yeah pretty important to add to the higher-cost workflows (though, I say that and we're using free runners, but, whatever)